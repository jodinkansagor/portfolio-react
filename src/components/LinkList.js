import React from 'react';
import Link from './Link';
import PropTypes from 'prop-types';
import styles from '../components/LinkList.css';

const LinkList = ({ linkList }) => {
  const linkElements = linkList.map((link, i) => (
    <li key={i}>
      <Link url={link.url} linkImg={link.linkImg} linkName={link.linkName} />
    </li>

  ));

  return (
    <ul>
      {linkElements}
    </ul>
  );
};

LinkList.propTypes = {
  linkList: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    linkImg: PropTypes.string.isRequired,
    linkName: PropTypes.string.isRequired
  })).isRequired
};

export default LinkList;
