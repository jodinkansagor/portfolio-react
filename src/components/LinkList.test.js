import React from 'react';
import { shallow } from 'enzyme';
import LinkList from './LinkList';

describe('LinkList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<LinkList linkList={['link', 'link']} />);
    expect(wrapper).toMatchSnapshot();
  });
});