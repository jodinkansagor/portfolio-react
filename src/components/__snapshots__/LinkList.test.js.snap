// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList component matches a snapshot 1`] = `
<ul>
  <li
    key="0"
  >
    <Link />
  </li>
  <li
    key="1"
  >
    <Link />
  </li>
</ul>
`;
